tests/integration/scheduling_test.go

```diff
@@ -0,0 +1,364 @@
+//go:build integration
+// +build integration
+
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"path"
+
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	sambaoperatorv1alpha1 "github.com/samba-in-kubernetes/samba-operator/api/v1alpha1"
+	"github.com/samba-in-kubernetes/samba-operator/tests/utils/kube"
+	//"github.com/samba-in-kubernetes/samba-operator/tests/utils/poll"
+	"github.com/samba-in-kubernetes/samba-operator/tests/utils/smbclient"
+)
+
+type CommonSelectorSuite struct {
+	suite.Suite
+
+	commonSources []kube.FileSource
+	testAuths     []smbclient.Auth
+	destNamespace string
+
+	// cached values
+	tc *kube.TestClient
+
+	// testID is a short unique test id, pseudo-randomly generated
+	testID string
+	// labeledNode has the special label applied to it
+	labeledNode string
+	// numPods to create for our tests
+	numPods int
+}
+
+func (s *CommonSelectorSuite) defaultContext() context.Context {
+	ctx := testContext()
+	if s.testID != "" {
+		ctx = context.WithValue(ctx, TestIDKey, s.testID)
+	}
+	return ctx
+}
+
+func (s *CommonSelectorSuite) getTestClient() *kube.TestClient {
+	return s.tc
+}
+
+func (s *CommonSelectorSuite) labelANode(ctx context.Context) {
+	// apply a special label to only one node.
+	// this test assumes that all linux-amd64 nodes in the test cluster can
+	// run a samba pod. If not, this test may fail in unexpected ways.
+	nodesList, err := s.tc.Clientset().CoreV1().Nodes().List(ctx, metav1.ListOptions{
+		LabelSelector: "kubernetes.io/os=linux,kubernetes.io/arch=amd64,!node-role.kubernetes.io/control-plane",
+	})
+	s.Require().NoError(err)
+	nodesAvail := len(nodesList.Items)
+	// did we find enough nodes for the test?
+	if nodesAvail < 2 {
+		// were we told we should have enough? Fail if we should have had enough
+		if nodesAvail < testMinNodeCount {
```

Comment:
With $NODE_COUNT defaulting to 3(from centosci script), above condition should have failed but its inside an earlier check for presence of minimum 2 worker nodes. So basically
if (2 < 2) {
        if (2 < 3) {
                ❌ FAIL ❌ 
        }
}

Nice 👍🏻